scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -220,37 +268,91 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
           next.is[CloseDelim] &&
           next.pos.startLine > curr.pos.endLine
       if (curr.isNot[Trivia] && !isTrailingComma) {
-        parserTokens += curr
-        parserTokenPositions += currPos
-        val sepRegions1 = {
-          if (curr.is[LeftParen]) ')' :: sepRegions
-          else if (curr.is[LeftBracket]) ']' :: sepRegions
+
+        var sepRegions = if (indentedRegion(sepRegionsP) && currentIndent >= 0) {
+          var sepRegionsI = sepRegionsP
+          while (indentedRegion(sepRegionsI) && sepRegionsI.head._2 > currentIndent) {
+            parserTokens += new Indentation.Outdent(curr.input, curr.dialect, curr.start, curr.end)
+            parserTokenPositions += currPos
+            parserTokens += new LF(curr.input, curr.dialect, curr.start)
+            parserTokenPositions += currPos
+            sepRegionsI = sepRegionsI.tail
+          }
+          sepRegionsI
+        } else sepRegionsP
+
+        sepRegions = if (addRegion > 0) {
+          if (!sepRegions.isEmpty && sepRegions.head._1 == '{') ('E', addRegion) :: sepRegions.tail
+          else ('O', addRegion) :: sepRegions
```

Comment:
i'd create a class, with named fields, instead of using the tuple.

Reply:
changed